From 91f095dbcf58bc3012c99285f808962a0460f12c Mon Sep 17 00:00:00 2001
From: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date: Tue, 16 May 2017 16:56:11 +0200
Subject: [PATCH] OS-6130 xhci and nvme can't bind DMA memory with IOMMU
 enabled Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by: Jerry
 Jelinek <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/uts/i86pc/io/rootnex.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/src/uts/i86pc/io/rootnex.c b/usr/src/uts/i86pc/io/rootnex.c
index d2faa09ef7..54f0b0f69e 100644
--- a/usr/src/uts/i86pc/io/rootnex.c
+++ b/usr/src/uts/i86pc/io/rootnex.c
@@ -2174,7 +2174,7 @@ out:
 	 */
 	if ((sinfo->si_copybuf_req == 0) &&
 	    (sinfo->si_sgl_size <= (unsigned)attr->dma_attr_sgllen) &&
-	    (dmao->dmao_size < dma->dp_maxxfer)) {
+	    (dmao->dmao_size <= dma->dp_maxxfer)) {
 fast:
 		/*
 		 * If the driver supports FMA, insert the handle in the FMA DMA
-- 
2.21.0

